script.js

```diff
@@ -1,4 +1,162 @@
+var GAME_STATE_CHOOSE_DICE_ORDER = "GAME_STATE_CHOOSE_DICE_ORDER";
+var GAME_STATE_COMPARE_SCORES = "GAME_STATE_COMPARE_SCORES";
+var gameState = GAME_STATE_DICE_ROLL;
+
+var currentPlayerRolls = [];
+
+var currentPlayer = 1;
+var allPlayersScore = [];
+
+// Helper Function
+var rollDice = function () {
+  console.log("Control flow: start of rollDice()");
+  // Random decimal between 0 and 6
+  var randomDecimal = Math.random() * 6;
+  // Random integer from 1 to 6
+  var randomInteger = Math.floor(randomDecimal) + 1;
+
+  console.log("rollDice output, randomInteger: ", randomInteger);
+  return randomInteger;
+};
+
+var rollDiceForPlayer = function () {
+  console.log("Control flow: start of rollDiceForPlayer()");
+  var counter = 0;
+  while (counter < 2) {
+    currentPlayerRolls.push(rollDice());
+    counter = counter + 1;
+  }
+
+  console.log(
+    "rollDiceForPlayer changes, currentPlayerRolls: ",
+    currentPlayerRolls
+  );
+  return (
+    "Welcome, Player " +
+    currentPlayer +
+    "<br><br>You rolled:<br>Dice 1: " +
+    currentPlayerRolls[0] +
+    " | Dice 2: " +
+    currentPlayerRolls[1] +
+    ".<br><br>Now, please input either '1' or '2' to choose the corresponding dice to be used as the first digit of your final value."
+  );
+};
+
+var getPlayerScore = function (playerInput) {
+  var playerScore;
+  // input validation
+  if (playerInput != 1 && playerInput != 2) {
+    console.log(
+      "Control flow: input validation, invalid inout... NOT 1 AND NOT 2"
+    );
+    return (
+      "Error! Please only input '1' or '' to choose which dice to use as the first digit.<br><br>Your dice rolls are:<br>Dice 1: " +
+      currentPlayerRolls[0] +
+      " | Dice 2: " +
+      currentPlayerRolls[1] +
+      "."
+    );
+  }
+  // input == 1
+  if (playerInput == 1) {
+    console.log("Control flow: input == 1");
+    playerScore = Number(
+      String(currentPlayerRolls[0]) + String(currentPlayerRolls[1])
+    );
+  }
+  // input == 2
+  else if (playerInput == 2) {
+    console.log("Control flow: input == 2");
+    playerScore = Number(
+      String(currentPlayerRolls[1]) + String(currentPlayerRolls[0])
+    );
+  }
+  // Store playerScore in array
+  allPlayersScore.push(playerScore);
+
+  // clear current player rolls array
+  currentPlayerRolls = [];
+  return "Player" + currentPlayer + ", your chosen value is: " + playerScore;
+};
+
+var comparePlayerScore = function () {
+  var compareMessage =
+    "Player 1 score: " +
+    allPlayersScore[0] +
+    "<br>Player 2 score: " +
+    allPlayersScore[1];
+
+  // player 1 wins
+  if (allPlayersScore[0] > allPlayersScore[1]) {
+    compareMessage = compareMessage + "<br><br>Player 1 wins!";
+  }
+  // player 2 wins
+  if (allPlayersScore[0] < allPlayersScore[1]) {
+    compareMessage = compareMessage + "<br><br>Player 2 wins!";
+  }
+  // tie
+  if (allPlayersScore[0] == allPlayersScore[1]) {
+    compareMessage = compareMessage + "<br><br>It's a tie!";
+  }
```

Comment:
slight nitpick here, it might be better to use:
```
if(...){
  // logic
} else if(...){
  // logic
} else {
  // logic
}
```

if/else if/else is better than consecutive ifs here.